storage/storage_image.go

```diff
@@ -341,16 +355,31 @@ func (s *storageImageDestination) Commit() error {
 	}
 	logrus.Debugf("created new image ID %q", img.ID)
 	s.ID = img.ID
+	names := []string{}
+	if s.imageRef.DockerReference() != nil {
+		names = append(names, verboseName(s.imageRef.DockerReference()))
+	}
 	if s.Tag != "" {
-		// We have a name to set, so move the name to this image.
-		if err := s.imageRef.transport.store.SetNames(img.ID, []string{s.Tag}); err != nil {
+		// Strip off any digest suffix, and if that leaves anything,
+		// use that name, too.
+		tag := s.Tag
+		if i := strings.Index(tag, "@"); i != -1 {
```

Comment:
Similar to above: AFAICT `s.Tag` comes from `s.imageRef.reference`, i.e. `refname` in `.ParseStoreReference()`, which is set based on `name` in the same function, and `name` is set based on `refInfo[0]` which comes from `SplitN(ref, "@", 2)`, i.e. `name` in there, and by extension `s.Tag`, can never contain a `@`.

Am I missing a code path to create a `storageReference `with different properties, perhaps?